discovery_engine_core/core/src/storage/sqlite.rs

```diff
@@ -84,113 +92,301 @@ impl Storage for SqliteStorage {
+        .into_iter()
+        .map(|doc| {
+            let url = Url::parse(&doc.url).map_err(|e| DatabaseError::TypeConversion(e.into()))?;
+            let image = doc
+                .image
+                .map(|url| Url::parse(&url).map_err(|e| DatabaseError::TypeConversion(e.into())))
+                .transpose()?;
+            let (lang_code, country_code) = doc.market.split_at(2);
+            let market = Market {
+                lang_code: lang_code.to_string(),
+                country_code: country_code.to_string(),
+            };
+
+            let news_resource = NewsResource {
+                title: doc.title,
+                snippet: doc.snippet,
+                topic: doc.topic,
+                url,
+                image,
+                date_published: doc.date_published,
+                source: doc.source,
+                market,
+            };
+            let newscatcher_data = NewscatcherData {
+                domain_rank: doc.domain_rank as usize,
+                score: Some(doc.score),
+            };
+            let user_reacted = doc.user_reaction.and_then(FromPrimitive::from_u32);
+
+            Ok(ApiDocumentView {
+                document: document::Id::from(doc.document),
+                news_resource,
+                newscatcher_data,
+                user_reacted,
+                in_batch_index: doc.in_batch_index as usize,
             })
-            .collect()
+        })
+        .collect()
+    }
+
+    async fn store_documents(
+        &self,
+        documents: &[document::Document],
+    ) -> Result<(), Self::FeedScopeError> {
+        if documents.is_empty() {
+            return Ok(());
+        }
+
+        // const BIND_LIMIT: usize = 32766;
+        // if documents.len() > BIND_LIMIT || documents.len() * 9 > BIND_LIMIT / 9 {
+        //     panic!();
+        // }
+
+        let mut tx = self.pool.begin().await.map_err(DatabaseError::Sql)?;
+        // Bulk inserts
+        // https://docs.rs/sqlx-core/latest/sqlx_core/query_builder/struct.QueryBuilder.html#method.push_values
+
+        // insert id into Document table (fk of HistoricDocument)
+        let mut query_builder = QueryBuilder::new("INSERT INTO Document (id) ");
+        query_builder.push_values(documents.iter(), |mut stm, doc| {
+            stm.push_bind(doc.id.as_uuid());
+        });
+        query_builder
+            .build()
+            .persistent(false)
+            .execute(&mut tx)
+            .await
+            .map_err(DatabaseError::Sql)?;
+
+        // insert id into HistoricDocument table
+        let mut query_builder = QueryBuilder::new("INSERT INTO HistoricDocument (document) ");
+        query_builder.push_values(documents.iter(), |mut stm, doc| {
+            stm.push_bind(doc.id.as_uuid());
+        });
+        query_builder
+            .build()
+            .persistent(false)
+            .execute(&mut tx)
+            .await
+            .map_err(DatabaseError::Sql)?;
```

Comment:
Not needed, there is a trigger which does this automatically for you.

The only reason there is no such trigger for Document is because the design long term plans to move
documents from stacks to the feed instead.